internal/plugin/grpc_client_plugin.go

```diff
@@ -79,14 +86,74 @@ func (gc *GrpcClient) Init(ctx context.Context, messagePipe bus.MessagePipeInter
 	gc.conn, err = grpc.DialContext(grpcClientCtx, serverAddr, agentGrpc.GetDialOptions(gc.config)...)
 	gc.connectionMutex.Unlock()
 
+	gc.fileServiceClient = v1.NewFileServiceClient(gc.conn)
+
 	if err != nil {
 		return err
 	}
 	backOffCtx, backoffCancel := context.WithTimeout(ctx, gc.config.Client.Timeout)
 
+	err = backoff.WaitUntil(backOffCtx, gc.config.Common, gc.createConnectionClient)
 	defer backoffCancel()
+	if err != nil {
+		return err
+	}
+
+	gc.subscribeMutex.Lock()
+	subscribeCtx, gc.subscribeCancel = context.WithCancel(ctx)
+	gc.subscribeMutex.Unlock()
 
-	return backoff.WaitUntil(backOffCtx, gc.config.Common, gc.createConnectionClient)
+	go gc.subscribe(subscribeCtx)
+
+	return nil
+}
+
+// wastedassign giving out that subscribeClient is set to nil
+// nolint: wastedassign
+func (gc *GrpcClient) subscribe(ctx context.Context) {
+	var subscribeClient v1.CommandService_SubscribeClient
+	var err error
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			subscribeClient, err = gc.commandServiceClient.Subscribe(ctx)
```

Comment:
Wait, aren't we resubscribing constantly now that this is inside the for loop? AFAIK, the subscribe call should be outside the for loop and we should be repeatedly calling `subscribeClient.Recv()` in the loop.

Reply:
very good spot, will fix it now